pkg/agent/client_test.go

```diff
@@ -354,6 +357,18 @@ func newClosePacket(connID int64) *client.Packet {
 	}
 }
 
+func waitForConnectionDeletion(t *testing.T, testClient *Client, connID int64) {
+	t.Helper()
+	var ok bool
+	wait.PollImmediate(100*time.Millisecond, 3*time.Second, func() (bool, error) {
```

Comment:
3 seconds is a pretty short timeout when running on over-scheduled CI infrastructure. Consider using a higher timeout.